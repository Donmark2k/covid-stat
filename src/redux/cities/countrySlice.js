import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const api = 'https://disease.sh/v3/covid-19/countries';

const initialState = {
  countryList: [],
  status: 'idle',
  error: null,
};

export const fetchCountries = createAsyncThunk('country/fetchCountries', async () => {
  try {
    const response = await axios.get(api);
    return response.data;
  } catch (error) {
    return error.message;
  }
});

export const countrySlice = createSlice({
  name: 'rockets',
  initialState,
  reducers: {
    reserveRocket: (state, payload) => ({
      ...state,
      rocketList: state.rocketList.map((rocket) => {
        if (rocket.id === payload.payload) {
          if (rocket.reserved === true) {
            return { ...rocket, reserved: false };
          }
          return { ...rocket, reserved: true };
        }
        return rocket;
      }),
    }),
  },
  extraReducers(builder) {
    builder.addCase(fetchRockets.pending, (state) => ({
      ...state,
      status: 'loading',
    }))
      .addCase(fetchRockets.fulfilled, (state, action) => ({
        ...state,
        rocketList: action.payload.map((rocket) => ({
          id: rocket.id,
          name: rocket.name,
          description: rocket.description,
          flickr_images: rocket.flickr_images[0],
        })),
        status: 'loaded',
      }))
      .addCase(fetchRockets.rejected, (state, action) => ({
        ...state,
        status: 'failed',
        error: [...state.error, action.error.message],
      }));
  },

});

export const { reserveRocket } = rocketSlice.actions;
export default rocketSlice.reducer;