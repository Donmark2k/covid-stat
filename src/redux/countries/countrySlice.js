import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const api = 'https://disease.sh/v3/covid-19/countries';

const initialState = {
  countryList: [],
  status: 'idle',
  error: null,
};

export const fetchCountries = createAsyncThunk('country/fetchCountries', async () => {
  try {
    const response = await axios.get(api);
    return response.data;
  } catch (error) {
    return error.message;
  }
});

export const countrySlice = createSlice({
  name: 'countries',
  initialState,
  reducers: {
    // reserveRocket: (state, payload) => ({
    //   ...state,
    //   rocketList: state.rocketList.map((rocket) => {
    //     if (rocket.id === payload.payload) {
    //       if (rocket.reserved === true) {
    //         return { ...rocket, reserved: false };
    //       }
    //       return { ...rocket, reserved: true };
    //     }
    //     return rocket;
    //   }),
    // }),
  },
  extraReducers(builder) {
    builder.addCase(fetchCountries.pending, (state) => ({
      ...state,
      status: 'loading',
    }))
      .addCase(fetchCountries.fulfilled, (state, action) => ({
        ...state,
        countryList: action.payload.map((country) => ({
          updated: country.id,
          country: country.name,
          cases: country.description,
          deaths: country.deaths,
        })),
        status: 'loaded',
      }))
      .addCase(fetchCountries.rejected, (state, action) => ({
        ...state,
        status: 'failed',
        error: [...state.error, action.error.message],
      }));
  },

});

export default countrySlice.reducer;