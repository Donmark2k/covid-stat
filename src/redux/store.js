// import { configureStore } from '@reduxjs/toolkit';
// import { createLogger } from 'redux-logger';
// import rocketReducer from './rockets/rocketSlice';
// import missionReducer from './missions/missionSlice';

// const logger = createLogger({
//   // Options for the logger can be passed here
//   // Show only the actions in the console
//   predicate: (getState, action) => action.type !== 'SOME_ACTION',
//   // Collapse the logs by default
//   collapsed: true,
// });

// export const store = configureStore({
//   reducer: {
//     missions: missionReducer,
//     rockets: rocketReducer,
//   },
//   middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
// });

// export default store;
