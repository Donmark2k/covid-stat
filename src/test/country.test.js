import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import axios from 'axios';
import Countries from '../components/country';
import { store } from '../redux/store';
import { fetchCountries } from '../redux/countries/countrySlice';

it('Country  page renders correctly', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <Countries />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// describe('Test for Country', () => {
//   test('should render', () => {
//     const country = {
//       id: 24,
//       country: 'Angola',
//       cases: '105298',
//     };
//     const tree = renderer
//        .create (
//       <Provider store={store}>
//         <Countries countries={country} key={24} />
//       </Provider>,
//     );
//     expect(tree).toMatchSnapshot();
//   });
// });

// Country  slicer and reducer check
describe('Countries redux state tests', () => {
  // it('Should initially hold country data by default', () => {
  //   const state = store.getState().countries;
  //   expect(state.countryList.length).toEqual(0);
  // });

  it('Rockets fetch data from API', async () => {
    const url = 'https://disease.sh/v3/covid-19/countries';
    const axiosSpy = jest.spyOn(axios, 'get');
    jest.setTimeout(90000);
    const dispatchSpy = jest.fn();
    await fetchCountries(url)(dispatchSpy);
    expect(axiosSpy).toHaveBeenCalledWith(url);
  });
});
