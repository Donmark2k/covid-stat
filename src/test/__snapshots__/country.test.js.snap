// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country  page renders correctly 1`] = `
<div
  className="container"
  style={
    Object {
      "display": "block",
      "margin": "auto",
      "width": "90%",
    }
  }
>
  <div
    className="row"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
  />
</div>
`;
